packages/neuron-ui/src/components/PageContainer/hooks.ts

```diff
@@ -0,0 +1,109 @@
+import { useCallback, useEffect, useState } from 'react'
+import { importedWalletDialogShown } from 'services/localCache'
+import { isDark, openExternal, updateWallet, setTheme as setThemeAPI } from 'services/remote'
+import { Migrate } from 'services/subjects'
+import { getExplorerUrl, isSuccessResponse } from 'utils'
+
+export const useSetBlockNumber = ({
+  firstAddress,
+  walletID,
+  isMainnet,
+  isLightClient,
+  isHomePage,
+}: {
+  firstAddress: string
+  walletID: string
+  isMainnet: boolean
+  isLightClient: boolean
+  isHomePage?: boolean
+}) => {
+  const [showSetStartBlock, setShowSetStartBlock] = useState(false)
+  const [startBlockNumber, setStartBlockNumber] = useState('')
+  const [blockNumberErr, setBlockNumberErr] = useState('')
+  const onChangeStartBlockNumber = useCallback((e: React.SyntheticEvent<HTMLInputElement>) => {
+    const { value } = e.currentTarget
+    const blockNumber = value.replace(/,/g, '')
+    if (Number.isNaN(+blockNumber) || /[^\d]/.test(blockNumber)) {
+      return
+    }
+    setStartBlockNumber(blockNumber)
+    setBlockNumberErr('')
+  }, [])
+  const onOpenAddressInExplorer = useCallback(() => {
+    const explorerUrl = getExplorerUrl(isMainnet)
+    openExternal(`${explorerUrl}/address/${firstAddress}`)
+  }, [firstAddress])
+  const onViewBlock = useCallback(() => {
+    const explorerUrl = getExplorerUrl(isMainnet)
+    openExternal(`${explorerUrl}/block/${startBlockNumber}`)
+  }, [startBlockNumber, isMainnet])
+  const onConfirm = useCallback(() => {
+    updateWallet({ id: walletID, startBlockNumberInLight: `0x${BigInt(startBlockNumber).toString(16)}` }).then(res => {
+      if (isSuccessResponse(res)) {
+        setShowSetStartBlock(false)
+      } else {
+        setBlockNumberErr(typeof res.message === 'string' ? res.message : res.message.content!)
+      }
+    })
+  }, [startBlockNumber, walletID])
+  const openDialog = useCallback(() => {
+    setShowSetStartBlock(true)
+    setStartBlockNumber('')
+  }, [])
+  useEffect(() => {
+    if (isHomePage) {
+      const needShow = importedWalletDialogShown.needShow(walletID)
+      if (needShow && isLightClient) {
+        openDialog()
+      }
```

Comment:
I have a question here.

If the wallet is imported when neuron is connected to a full node, will it be set along with `start block number`?

Reply:
It will not show the setting dialog.

Reply:
Then how can I set the start point when neuron is switched to the light client mode?

Maybe a check should be added on network switching

Reply:
They can set the start block number here.

<img width="487" alt="image" src="https://github.com/nervosnetwork/neuron/assets/12881040/8fc64911-1a36-4b33-8c67-b19bea3338f7">

Reply:
Gotcha